### PR TITLE
kubernetes: Migrate to v1beta3 based Kubernetes

### PR DESCRIPTION
**Pull request by stefwalter**
* Currently a lot of stuff is hard coded to the default namespace
 * Renamed 'minion' -> 'node' as done in the v1beta3 API
 * Renamed 'replicationControllers' -> 'replicationcontrollers' as done
   in the v1beta3 API

### CODE REVIEW

**Comment by stefwalter** [comment]
@sub-mod Could you test and/or review this pull request?

**Comment by stefwalter** [comment]
Rebased onto master.

File: pkg/kubernetes/examples/database-service.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "kind": "Service",
+  "apiVersion": "v1beta3",
+
+	"metadata": {
+		"name": "database",
+		"namespace": "default"
```
Comment (mvollmer) [inline pkg/kubernetes/examples/database-service.json]: Comma?

File: pkg/kubernetes/kubernetes.js [hunk 2]
```diff
@@ -129,93 +129,93 @@ define([
...
-        this.create_minion = function create_minion(minion_json) {
-            api.post("/api/v1beta1/minions", minion_json)
+        this.create_node = function create_node(node_json) {
+            api.post("/api/v1beta3/nodes", node_json)
                .fail(failure);
         };
 
         this.create_pod = function create_pod(pod_json) {
-            api.post("/api/v1beta1/pods", pod_json)
+            api.post("/api/v1beta3/pods", pod_json)
```
Comment (sub-mod) [inline pkg/kubernetes/kubernetes.js]: should be api/v1beta3/namespaces/default/pods

File: pkg/kubernetes/kubernetes.js [hunk 3]
```diff
@@ -129,93 +129,93 @@ define([
...
                .fail(failure);
         };
 
         this.create_service = function create_pod(service_json) {
-            api.post("/api/v1beta1/services", service_json)
+            api.post("/api/v1beta3/services", service_json)
```
Comment (sub-mod) [inline pkg/kubernetes/kubernetes.js]: should be api/v1beta3/namespaces/default/services

File: pkg/kubernetes/kubernetes.js [hunk 4]
```diff
@@ -129,93 +129,93 @@ define([
...
                 "body": "",
-                "path": "/api/v1beta1/services/" + encodeURIComponent(service_name)
+                "path": "/api/v1beta3/namespaces/default/services/" + encodeURIComponent(service_name)
             }).fail(failure);
         };
 
-        this.create_replicationController = function create_replicationController(replicationController_json) {
-            api.post("/api/v1beta1/replicationControllers", replicationController_json)
+        this.create_replicationcontroller = function create_replicationcontroller(replicationcontroller_json) {
+            api.post("/api/v1beta3/replicationcontrollers", replicationcontroller_json)
```
Comment (sub-mod) [inline pkg/kubernetes/kubernetes.js]: should be api/v1beta3/namespaces/default/replicationcontrollers

**Comment by stefwalter** [comment]
@mvollmer @sub-mod Thanks for catching those issues. Fixed, and pushed an amended commit.

**Pull request closed without merging**